### PR TITLE
feat(map): add a directive that change url to path depending on the network status

### PR DESCRIPTION
**Pull request by drekss**
* new directive that change url to path if network is off and change path to url if network is on.
* query can now get features from layers.

### CODE REVIEW

File: packages/geo/src/lib/datasource/shared/datasources/datasource.ts [hunk 1]
```diff
@@ -31,5 +33,5 @@ export abstract class DataSource {
     return this.options.legend ? [this.options.legend] : [];
   }
 
-  protected abstract onUnwatch();
+  onLayerStatusChange(status: SubjectStatus): void {}
```
Comment (mbarbeau) [inline packages/geo/src/lib/datasource/shared/datasources/datasource.ts]: ??

File: packages/geo/src/lib/map/map.module.ts [hunk 2]
```diff
@@ -24,23 +25,25 @@ import { MiniBaseMapComponent } from './baselayers-switcher/mini-basemap.compone
     IgoConfirmDialogModule,
     MatIconModule,
     MatButtonModule,
-    MatTooltipModule
+    MatTooltipModule,
```
Comment (mbarbeau) [inline packages/geo/src/lib/map/map.module.ts]: useless comma

File: packages/geo/src/lib/map/shared/mapLayer.directive.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import { IgoMap } from './map';
+import { MapBrowserComponent } from '../map-browser/map-browser.component';
+import { NetworkService, ConnectionState } from '@igo2/core';
+import { Subscription } from 'rxjs';
+import { Layer } from '../../layer/shared/layers/layer';
+
+@Directive({
+    selector: '[igoMapLayer]'
+  })
+export class MapLayerDirective implements AfterViewInit {
```
Comment (mbarbeau) [inline packages/geo/src/lib/map/shared/mapLayer.directive.ts]: The name of the class is not significant

File: packages/geo/src/lib/query/shared/query.directive.ts [hunk 4]
```diff
@@ -56,7 +56,7 @@ export class QueryDirective implements AfterViewInit, OnDestroy {
   /**
    * Whter to query features or not
    */
-  @Input() queryFeatures: boolean = false;
+  @Input() queryFeatures: boolean = true;
```
Comment (mbarbeau) [inline packages/geo/src/lib/query/shared/query.directive.ts]: Explanation of the change ... should not be changed on the component?

File: packages/geo/src/lib/query/shared/query.directive.ts [hunk 5]
```diff
@@ -66,7 +66,7 @@ export class QueryDirective implements AfterViewInit, OnDestroy {
   /**
    * Feature query hit tolerance
    */
-  @Input() queryFeaturesCondition: (olLayer: OlLayer) => boolean;
+  @Input() queryFeaturesCondition: (olLayer: OlLayer) => true;
```
Comment (mbarbeau) [inline packages/geo/src/lib/query/shared/query.directive.ts]: ????

File: packages/geo/src/lib/datasource/shared/datasources/datasource.interface.ts [hunk 6]
```diff
@@ -25,6 +25,8 @@ export interface DataSourceOptions {
   // TODO: Should those options really belong here?
   sourceFields?: SourceFieldsOptionsParams[];
   download?: DownloadOptions;
+  url?: string;
```
Comment (mbarbeau) [inline packages/geo/src/lib/datasource/shared/datasources/datasource.interface.ts]: Should be put only on the types of datasources supporting an offline url.

Comment (drekss) [reply packages/geo/src/lib/datasource/shared/datasources/datasource.interface.ts]: the sourceOptions layer from the map is attributed to dataSourceOptions and not from the other datasource type. I can't can get url or path from a datasource if url is not in dataSourceOptions

File: packages/geo/src/lib/map/shared/mapLayer.directive.ts (hunk 3 above)
Comment (drekss) [reply packages/geo/src/lib/map/shared/mapLayer.directive.ts]: is igoChangeMapSourceUrl would be ok?

Comment (mbarbeau) [reply packages/geo/src/lib/map/shared/mapLayer.directive.ts]: MapLayerOfflineDirective ?
MapOfflineDirective ?

**Pull request closed without merging**